src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let memory = [];
+
+  const adder = (number) => {
+    if (number === undefined) {
+      const sum = memory.reduce((acc, curr) => acc + curr, 0);
+
+      memory = [];
+
+      return sum;
+    } else {
+      memory.push(number);
```

Comment:
Instead of pushing the number to an array, you could add it directly to a running total, which would be more efficient.